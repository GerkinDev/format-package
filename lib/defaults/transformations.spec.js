jest.mock('sort-scripts');

const sortScripts = require('sort-scripts');
const transformations = require('./transformations');

describe('default transformations', () => {
  describe('scripts', () => {
    beforeEach(() => {
      sortScripts.mockReset();
      sortScripts.mockReturnValue(
        new Array(5).fill(0).map((_, i) => [`name${i}`, `value${i}`])
      );
    });

    it('sorts the scripts based on 3rd party module', () => {
      expect(transformations.scripts('key', 'value')).toMatchSnapshot();
      expect(sortScripts).toHaveBeenCalledWith('value');
    });
  });
});
