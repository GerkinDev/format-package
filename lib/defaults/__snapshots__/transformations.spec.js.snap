// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default transformations scripts sorts the scripts based on 3rd party module 1`] = `
Array [
  "key",
  Object {
    "name0": "value0",
    "name1": "value1",
    "name2": "value2",
    "name3": "value3",
    "name4": "value4",
  },
]
`;
